### ISSUE TITLE
Long "search" time

### ISSUE THREAD

**Issue by RarePotato8DE** [Tyrrrz/YoutubeExplode#412]
Hey, first off I have to thank you for releasing and creating YouTubeExplode. It's awesome :)

So I started a little project with the SDK which is basically what you are already offering: Downloading music (videos).
Yesterday everything was quick and downloading/ getting the video or playlist took a few seconds.

Now I noticed it takes way longer. I've added a stopwatch which tells me how long something took (getting the video and downloading). Yesterday downloading of a playlist with 50 videos (and converting to .mp3 with your YouTubeConvert) took 18 minutes which is about 22 seconds per video.

Now just getting a video via. #.Videos.GetAsync takes 60 seconds and downloading the video also takes another 50 seconds.
I definitly know it has nothing to do with your Project, so creating an "issue" is wrong. But do you have any idea why that could happen?

I wasn't sure if I somehow messed something up, so I tried your examples and they take also a long time.

Thanks in advance.
Keep this project up :)

Kind regards.

**Comment by Tyrrrz** [comment]
Hi
Honestly, there could be a range of factors. To streamline things a bit, you can iterate over videos instead of getting all of them and download them one by one.

**Comment by RarePotato8DE** [comment]
Hey,
thanks for responding.

What you said it is for full playlist downloading. But as I said it even takes ~2min for a single .mp3 download. Before it was 22seconds.
I just tried it now, and getting a video just takes 2 seconds and downloading 12 secoonds (all depends on video lenght, but I've used for my examples almost the same video lenghts, ~3 mins).
There were probably some server issues or even my connection (altough I've checked it yesterday and it was normal).

My "issue" is solved. 
Kind regards and thanks for your work on "YouTubeExplode" :)
